lib/content-services/src/lib/search-ai/services/search-ai.service.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * @license
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { AiAnswerPaging, QuestionModel, QuestionRequest, SearchAiApi } from '@alfresco/js-api';
+import { AlfrescoApiService } from '@alfresco/adf-core';
+import { BehaviorSubject, from, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { SelectionState } from '@alfresco/adf-extensions';
+import { TranslateService } from '@ngx-translate/core';
+import { SearchAiInputState } from '../models/search-ai-input-state';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SearchAiService {
+    private toggleSearchAiInput = new BehaviorSubject<SearchAiInputState>({
+        active: false
+    });
+    private readonly textFileMimeTypes = [
+        'application/msword',
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+        'application/vnd.oasis.opendocument.text',
+        'application/rtf',
+        'text/plain',
+        'application/pdf'
+    ];
+    private _searchAiApi: SearchAiApi;
+
+    get searchAiApi(): SearchAiApi {
+        this._searchAiApi = this._searchAiApi ?? new SearchAiApi(this.apiService.getInstance());
+        return this._searchAiApi;
+    }
+
+    toggleSearchAiInput$ = this.toggleSearchAiInput.asObservable();
+
+    constructor(private apiService: AlfrescoApiService, private translateService: TranslateService) {}
+
+    /**
+     * Update the state of the search AI input.
+     *
+     * @param state The new state of the search AI input.
+     */
+    updateSearchAiInputState(state: SearchAiInputState): void {
+        this.toggleSearchAiInput.next(state);
+    }
+
+    /**
+     * Ask a question to the AI.
+     *
+     * @param question The question to ask.
+     * @param mocked temporary parameter to mock returned information about question. Should be removed when backend implemented.
+     * @returns QuestionModel object containing information about questions.
+     */
+    ask(question: QuestionRequest, mocked = true): Observable<QuestionModel> {
```

Comment:
better have a `mocked` as a property of the service instead of a public parameter on each api, or we can have a clean implementation of the service and then replace it with the Mock instance at the app level

Reply:
done, changed it to property